astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -16,16 +15,16 @@
 
 from .cirs import CIRS
 from .altaz import AltAz
+from .hadec import HADec
 from .utils import PIOVER2
 from ..erfa_astrom import erfa_astrom
 
 
-@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, CIRS, AltAz)
-def cirs_to_altaz(cirs_coo, altaz_frame):
-    if (np.any(altaz_frame.location != cirs_coo.location) or
-            np.any(cirs_coo.obstime != altaz_frame.obstime)):
-        cirs_coo = cirs_coo.transform_to(CIRS(obstime=altaz_frame.obstime,
-                                              location=altaz_frame.location))
+def cirs_to_observed(cirs_coo, observed_frame, coord_type):
```

Comment:
In principle, we do not really need `coord_type`, since it is implicit in the frame (`coord_type = 'A' if insinstance(observed_frame, AltAz) else 'H'`). With that argument removed, the same function could serve both types of transforms directly. 

Though not super important.

Reply:
I like this idea. Will do.